unitary/quantum_chess/circuit_transformer.py

```diff
@@ -400,48 +400,43 @@ def transform(self, circuit: cirq.Circuit) -> cirq.Circuit:
+            return controlled_iswap.controlled_sqrt_iswap(*qubits, *op.controls)
+        if op.gate.sub_gate == cirq.ISWAP**-0.5:
+            return controlled_iswap.controlled_inv_sqrt_iswap(*qubits, *op.controls)
+    return op
+
+
+@cirq.transformer
+def decompose_into_sqrt_iswap(
+    circuit: cirq.AbstractCircuit, *, context: Optional[cirq.TransformerContext] = None
+) -> cirq.Circuit:
+    """Transformer that decomposes all three qubit operations into
     sqrt-ISWAPs.
 
     Currently supported are controlled ISWAPs with a single control
     and control-X gates with multiple controls (TOFFOLI gates).:w
     """
-
-    def optimization_at(
-        self, circuit: cirq.Circuit, index: int, op: cirq.Operation
-    ) -> Optional[cirq.PointOptimizationSummary]:
-        if len(op.qubits) > 3:
-            raise DeviceMappingError(f"Four qubit ops not yet supported: {op}")
-        new_ops = None
-        if op.gate == cirq.SWAP or op.gate == cirq.CNOT or op.gate == cirq.TOFFOLI:
-            new_ops = cg.optimized_for_sycamore(cirq.Circuit(op))
-        elif isinstance(op, cirq.ControlledOperation):
-            if not all(v == 1 for values in op.control_values for v in values):
-                raise DeviceMappingError(f"0-controlled ops not yet supported: {op}")
-            qubits = op.sub_operation.qubits
-            if op.gate.sub_gate == cirq.ISWAP:
-                new_ops = controlled_iswap.controlled_iswap(*qubits, *op.controls)
-            if op.gate.sub_gate == cirq.ISWAP**-1:
-                new_ops = controlled_iswap.controlled_iswap(
-                    *qubits, *op.controls, inverse=True
-                )
-            if op.gate.sub_gate == cirq.ISWAP**0.5:
-                new_ops = controlled_iswap.controlled_sqrt_iswap(*qubits, *op.controls)
-            if op.gate.sub_gate == cirq.ISWAP**-0.5:
-                new_ops = controlled_iswap.controlled_inv_sqrt_iswap(
-                    *qubits, *op.controls
-                )
-            if op.gate.sub_gate == cirq.X:
-                if len(op.qubits) == 2:
-                    new_ops = cg.optimized_for_sycamore(
-                        cirq.Circuit(cirq.CNOT(*op.controls, *qubits))
-                    )
-                if len(op.qubits) == 3:
-                    new_ops = cg.optimized_for_sycamore(
-                        cirq.Circuit(cirq.TOFFOLI(*op.controls, *qubits))
-                    )
-        if new_ops:
-            return cirq.PointOptimizationSummary(
-                clear_span=1, clear_qubits=op.qubits, new_operations=new_ops
-            )
+    if context is None:
+        context = cirq.TransformerContext()
+    converted_iswap_circuit = cirq.map_operations_and_unroll(
+        circuit.unfreeze(False),
+        transform_controlled_iswap,
+        deep=True,
+        tags_to_ignore=context.tags_to_ignore,
+    )
+    return cirq.optimize_for_target_gateset(
+        converted_iswap_circuit, gateset=SQRT_ISWAP_GATESET
+    )
```

Comment:
Pass the `context=context` so that `deep=True` is propagated in all the nested transformers as well.